rllib/evaluation/collectors/simple_list_collector.py

```diff
@@ -232,131 +234,69 @@ def build(self, view_requirements: ViewRequirementsDict) -> SampleBatch:
-
-                # Batch repeat (only provide a value every n timesteps).
-                if view_req.batch_repeat_value > 1:
-                    count = int(
-                        math.ceil(
-                            (len(np_data[data_col][0]) - self.shift_before)
-                            / view_req.batch_repeat_value
-                        )
-                    )
-                    data = [
-                        np.asarray(
-                            [
-                                d[
-                                    self.shift_before
-                                    + (i * view_req.batch_repeat_value)
-                                    + shift
-                                ]
-                                for i in range(count)
-                            ]
-                        )
-                        for d in np_data[data_col]
-                    ]
-                # Shift is exactly 0: Use trajectory as is.
-                elif shift == 0:
-                    data = [d[self.shift_before :] for d in np_data[data_col]]
-                # Shift is positive: We still need to 0-pad at the end.
-                elif shift > 0:
-                    data = [
-                        _to_float_np_array(
-                            np.concatenate(
-                                [
-                                    d[self.shift_before + shift :],
-                                    [
-                                        np.zeros(
-                                            shape=view_req.space.shape,
-                                            dtype=view_req.space.dtype,
-                                        )
-                                        for _ in range(shift)
-                                    ],
-                                ]
-                            )
-                        )
-                        for d in np_data[data_col]
-                    ]
-                # Shift is negative: Shift into the already existing and
-                # 0-padded "before" area of our buffers.
+
+                    # handle the case where the inds are out of bounds from the end.
+                    # if during the indexing any of the indices are out of bounds, we
+                    # need to use padding on the end to fill in the missing indices.
+                    element_at_t = []
+                    for index in inds:
+                        if index < len(d):
+                            element_at_t.append(d[index])
+                        else:
+                            # zero pad similar to the last element.
+                            element_at_t.append(zeros_like(d[-1]))
+                    element_at_t = np.stack(element_at_t)
+
+                    if element_at_t.shape[0] == 1:
+                        # squeeze to remove the T dimension if it is 1.
+                        element_at_t = element_at_t.squeeze(0)
```

Comment:
is there a unit test to cover this?? maybe add a TODO if there is not?

Reply:
implicitly yes. I do some shape checking in agent_collector unittest that rely on this line of code. For example, in `test_shift_by_one_with_repeat_value_larger_1`, `self.assertEqual(sample_batch["prev_obses"].shape, (2, 4))` if you don't do this the returned shape would be (2, 1, 4) I think.